### ISSUE TITLE
Uncaught error with _openssl.so with certain commands

### ISSUE THREAD

**Issue by justinmchase** [Azure/azure-cli#4616]
### Description
Full error and command here:
```
$ az webapp deployment user set --user-name example --password foobar
/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/cryptography/hazmat/bindings/_openssl.so: cannot enable executable stack as shared object requires: Invalid argument
Traceback (most recent call last):
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/azure/cli/main.py", line 36, in main
    cmd_result = APPLICATION.execute(args)
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/azure/cli/core/application.py", line 176, in execute
    self.configuration.load_params(command)
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/azure/cli/core/application.py", line 50, in load_params
    commands.load_params(command)
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/azure/cli/core/commands/__init__.py", line 391, in load_params
    command_table[command].load_arguments()
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/azure/cli/core/commands/__init__.py", line 335, in load_arguments
    self.arguments.update(self.arguments_loader())
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/azure/cli/core/commands/__init__.py", line 653, in arguments_loader
    return extract_args_from_signature(get_op_handler(operation), no_wait_param=no_wait_param)
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/azure/cli/core/commands/__init__.py", line 537, in get_op_handler
    op = import_module(mod_to_import)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/azure/cli/command_modules/appservice/custom.py", line 12, in <module>
    import OpenSSL.crypto
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/OpenSSL/__init__.py", line 8, in <module>
    from OpenSSL import crypto, SSL
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/OpenSSL/crypto.py", line 16, in <module>
    from OpenSSL._util import (
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/OpenSSL/_util.py", line 6, in <module>
    from cryptography.hazmat.bindings.openssl.binding import Binding
  File "/home/justin/lib/azure-cli/local/lib/python2.7/site-packages/cryptography/hazmat/bindings/openssl/binding.py", line 13, in <module>
    from cryptography.hazmat.bindings._openssl import ffi, lib
ImportError: /home/justin/lib/azure-cli/local/lib/python2.7/site-packages/cryptography/hazmat/bindings/_openssl.so: cannot enable executable stack as shared object requires: Invalid argument
```

---

### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
I followed the windows install steps here:
https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
```
21:03:15:justin:/mnt/e/code/justinmchase-example$ az -v
azure-cli (2.0.18)

acr (2.0.12)
acs (2.0.16)
appservice (0.1.17)
backup (1.0.1)
batch (3.1.4)
billing (0.1.5)
cdn (0.0.9)
cloud (2.0.8)
cognitiveservices (0.1.8)
command-modules-nspkg (2.0.1)
component (2.0.7)
configure (2.0.11)
consumption (0.1.5)
container (0.1.11)
core (2.0.17)
cosmosdb (0.1.13)
dla (0.0.12)
dls (0.0.14)
eventgrid (0.1.4)
extension (0.0.3)
feedback (2.0.6)
find (0.2.7)
interactive (0.3.10)
iot (0.1.12)
keyvault (2.0.11)
lab (0.0.11)
monitor (0.0.10)
network (2.0.15)
nspkg (3.0.1)
profile (2.0.13)
rdbms (0.0.7)
redis (0.2.9)
resource (2.0.15)
role (2.0.12)
servicefabric (0.0.4)
sql (2.0.12)
storage (2.0.16)
vm (2.0.15)

Python location '/home/justin/lib/azure-cli/bin/python'
Extensions directory '/home/justin/.azure/cliextensions'

Python (Linux) 2.7.6 (default, Oct 26 2016, 20:30:19)
[GCC 4.8.4]
```

**OS Version:** What OS and version are you using?  
Windows 10

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Bash on Ubuntu on Windows

**Comment by justinmchase** [comment]
Based on some other solutions to this I've seen I got a little further:
```
$ sudo apt-get install execstack
$ sudo execstack -c /home/justin/lib/azure-cli/local/lib/python2.7/site-packages/cryptography/hazmat/bindings/_openssl.so
$ az webapp deployment user set --user-name justin --password foobar
Operation failed with status: 'Bad Request'. Details: 400 Client Error: Bad Request for url: https://management.azure.com/providers/Microsoft.Web/publishingUsers/web?api-version=2016-03-01
```
I am wondering if I have something wrong with my openssl install or python? I installed python27 much earlier than this for use with nodejs. Which version of python should be used by this sdk?

**Comment by justinmchase** [comment]
I tried this:
```
pip install --upgrade --user 'cryptography==1.9'
```
But no dice.

**Comment by derekbekoe** [comment]
Have you seen https://github.com/Azure/azure-cli/issues/4154?
I created a PR to fix this in Bash on Windows with the apt-get install https://github.com/Azure/azure-cli/pull/4251/files.

Also, how did you install as looking at the install path `/home/justin/lib/azure-cli/bin/python`, it's not installed with apt-get which is recommended on Bash on Windows.

**Comment by justinmchase** [comment]
Ah, that is a very helpful clue.

The first time I tried to install I ran this via GitBash:
```
curl -L https://aka.ms/InstallAzureCli | bash
```

After struggling with that for a while, I was informed that that wasn't supported and I should be using Bash on Windows + `apt-get` instead. So it looks like I have some cruft in my environment that I need to clean out that I wasn't aware was bleeding over. I'll do that and re-update in Bash on Windows.

Regarding that PR you linked, thanks again. I'm ok with running that manually for now and look forward to it rolling out in the official package update. This should unblock me, if I hit any other issues I'll open separate tickets thanks.
